lang/cs/Org.Apache.REEF.Driver/Bridge/Events/ClosedContext.cs

```diff
@@ -40,52 +39,35 @@ public ClosedContext(IClosedContextClr2Java clr2java)
         }
 
         [DataMember]
-        public string InstanceId { get; set; }
+        public string InstanceId { get; private set; }
+
+        private IActiveContextClr2Java ParentContextClr2Java { get; set; }
+        private IClosedContextClr2Java ClosedContextClr2JavaClr2Java { get; set; }
 
         public string Id
         {
-            get
-            {
-                return _id;
-            }
+            get { return _id; }
         }
 
         public string EvaluatorId
         {
-            get
-            {
-                return _evaluatorId;
-            }
-
-            set
-            {
-            }
+            get { return _evaluatorId; }
         }
 
-        public Optional<string> ParentId { get; set; }
+        public Optional<string> ParentId
+        {
+            // TODO[REEF-762]: Implement
+            get { return Optional<string>.Empty(); }
+        }
 
         public IEvaluatorDescriptor EvaluatorDescriptor
         {
-            get
-            {
-                return ClosedContextClr2JavaClr2Java.GetEvaluatorDescriptor();
-            }
-
-            set
-            {
-            }
+            get { return ClosedContextClr2JavaClr2Java.GetEvaluatorDescriptor(); }
         }
 
         public IActiveContext ParentContext
         {
-            get
-            {
-                return new ActiveContext(ParentContextClr2Java);
-            }
+            get { return new ActiveContext(ParentContextClr2Java); }
```

Comment:
Yes, but we don't know when `ParentContextClr2Java` will be populated. Hence, I'd prefer not to change the logic here at this time.